internal/provider/resources/deployment.go

```diff
@@ -238,6 +239,7 @@ func (r *DeploymentResource) Schema(_ context.Context, _ resource.SchemaRequest,
 				Optional:    true,
 				Computed:    true,
 				CustomType:  jsontypes.NormalizedType{},
+				Default:     stringdefault.StaticString("{}"),
```

Comment:
so with `job_variable`, `parameter`, and `paramater_openai_schema` -- these are `jsontypes.NormalizedType{}`

but for the deployment_access tests, where we have HCL like this:

```
resource "prefect_deployment" "test" {
  name = "my-deployment"
  flow_id = prefect_flow.test.id
}
```

i was getting errors like this:

```
│ Error: Normalized JSON Unmarshal Error
│
│   with prefect_deployment.test,
│   on prefect_deployment.tf line 47, in resource "prefect_deployment" "test":
│   47: resource "prefect_deployment" "test" {
│
│ json string value is unknown
```

i _**think**_ what happened here was - when we swapped from req.Config -> req.Plan, we're now using the computed HCL configuration in the golang code.  So before, we're using req.Config, I want to say that it just treated unset values as `null`.  But when you actually use the resource schema via req.Plan, they become `unknown` if unset - the expectation is that the API may set something here

so when the error says "json string value is unknown", it's attempting to set what is now an unknown value from the computed plan -> golang struct that expects a json.NormalizedType

Because we want to make these an optional field, the fix here is to set a default value

Reply:
Similar to comment below, I wonder if we need to be making better use of `IsNull()` here - because I would have avoided setting a default because I assumed it'd try to always send that field in the payload. But makes sense that if the default value looks "empty" to the `json` package, it should `omitempty` it.

Reply:
>But makes sense that if the default value looks "empty" to the json package, it should omitempty it.

or maybe what this Default does is that it always sends the empty JSON object?

Reply:
🤔 I'll have to read up on this a lot more